src/System.IO/src/System/IO/TextWriter.cs

```diff
@@ -597,5 +598,332 @@ public override void Write(char value)
+            {
+                _out.WriteLine(buffer, index, count);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(bool value)
+            {
+                _out.WriteLine(value);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(int value)
+            {
+                _out.WriteLine(value);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(uint value)
+            {
+                _out.WriteLine(value);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(long value)
+            {
+                _out.WriteLine(value);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(ulong value)
+            {
+                _out.WriteLine(value);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(float value)
+            {
+                _out.WriteLine(value);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(double value)
+            {
+                _out.WriteLine(value);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(string value)
+            {
+                _out.WriteLine(value);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(object value)
+            {
+                _out.WriteLine(value);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(string format, object arg0)
+            {
+                _out.WriteLine(format, arg0);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(string format, object arg0, object arg1)
+            {
+                _out.WriteLine(format, arg0, arg1);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(string format, object arg0, object arg1, object arg2)
+            {
+                _out.WriteLine(format, arg0, arg1, arg2);
+            }
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override void WriteLine(string format, object[] arg)
+            {
+                _out.WriteLine(format, arg);
+            }
+
+            //
+            // On SyncTextWriter all APIs should run synchronously, even the async ones.
+            //
+
+            [MethodImpl(MethodImplOptions.Synchronized)]
+            public override Task WriteAsync(char value)
+            {
+                Write(value);
```

Comment:
Similar exceptions-escape-from-this-but-we-should-probably-leave-the-behavior-matching-desktop comment :(